packages/openapi-generator/src/document-converter.ts

```diff
@@ -75,47 +75,32 @@ export async function convertDocToOpenApiV3(
  * @param openApiDocument OpenAPI JSON document.
  * @returns The modified document.
  */
-export function convertDocWithDefaultTag(
+export function convertDocWithApiNameTag(
   openApiDocument: OpenAPIV3.Document
 ): OpenAPIV3.Document {
-  const tag = 'default';
+  const defaultTag = 'default';
 
-  executeForAllOperationObjects(openApiDocument, operation => {
-    operation.tags = addGlobalTagToOperationWhenNoTagsAreUsed(
+  executeForAllOperationObjects(openApiDocument, (operation,path, method, extensionApiName) => {
+    operation.tags = extensionApiName? [extensionApiName]:addGlobalTagToOperationWhenNoTagsAreUsed(
```

Comment:
The way the tags are created here, seems slightly more complex than necessary. `addGlobalTagToOperationWhenNoTagsAreUsed` is only used here and would be easier to read if combined with the code here. Consider adding something like `getTags(operation.tags, extensionApiName)`.